Mathlib/Algebra/Category/ModuleCat/ChangeOfRings.lean

```diff
@@ -76,24 +77,19 @@ def restrictScalars {R : Type u₁} {S : Type u₂} [Ring R] [Ring S] (f : R →
     ModuleCat.{v} S ⥤ ModuleCat.{v} R where
   obj := RestrictScalars.obj' f
   map := RestrictScalars.map' f
-  map_id _ := LinearMap.ext fun _ => rfl
-  map_comp _ _ := LinearMap.ext fun _ => rfl
 
 instance {R : Type u₁} {S : Type u₂} [Ring R] [Ring S] (f : R →+* S) :
     (restrictScalars.{v} f).Faithful where
-  map_injective h :=
-    LinearMap.ext fun x => by simpa only using DFunLike.congr_fun h x
+  map_injective h := by
+    ext x
+    simpa only using DFunLike.congr_fun (ModuleCat.hom_ext_iff.mp h) x
 
 instance {R : Type u₁} {S : Type u₂} [Ring R] [Ring S] (f : R →+* S) :
     (restrictScalars.{v} f).PreservesMonomorphisms where
   preserves _ h := by rwa [mono_iff_injective] at h ⊢
 
 -- Porting note: this should be automatic
-instance {R : Type u₁} {S : Type u₂} [Ring R] [Ring S] {f : R →+* S}
-    {M : ModuleCat.{v} S} : Module R <| (restrictScalars f).obj M :=
-  inferInstanceAs <| Module R <| RestrictScalars.obj' f M
-
--- Porting note: this should be automatic
+-- TODO: this instance gives diamonds if `f : S →+* S`...
```

Comment:
This instance causes downstream issues in `Mathlib/Algebra/Category/ModuleCat/Presheaf/Pushforward.lean:40`, where the synthesized instance does not equal the one obtained by unification.